src/plugins/analysis/software_components/internal/install.py

```diff
@@ -0,0 +1,24 @@
+import os
+import sys
+
+THIS_FILE = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(THIS_FILE, '', 'internal'))
+from scan_signatures import get_scanned_software
+
+
+def scan():
+    scanned_software = get_scanned_software('./signatures/os.yara')
```

Comment:
Please do not mix up string formating and read/write operations in one function.

Reply:
you might want to use common_helper_files.write_binary_to_file function to avoid errors and create folders automatically